kernel/src/device/tdxguest/mod.rs

```diff
@@ -176,36 +221,40 @@ fn handle_get_report(arg: usize) -> Result<i32> {
     let report = tdx_get_report(&user_request.report_data)?;
 
     let tdx_report_vaddr = arg + TDX_REPORTDATA_LEN;
-    user_space.write_bytes(tdx_report_vaddr, &mut VmReader::from(report.as_ref()))?;
+    user_space.write_bytes(tdx_report_vaddr, report.reader().limit(TDX_REPORT_LEN))?;
```

Comment:
I am wondering if `TDX_REPORT_LEN == size_of::<TdReport>()`. If so, use the latter one to draw a connection between the struct definition and the copy length. If not, this could be surprising and need some clarifications.